src/Filter/ToE164.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Laminas\I18n\PhoneNumber\Filter;
+
+/**
+ * @psalm-template T
+ */
+final class ToE164 extends AbstractFilter
+{
+    /**
+     * Given a valid phone number, return the number in E164 format, otherwise return the unfiltered input
+     *
+     * @param T $value
+     * @return string|T
+     */
+    public function filter($value)
+    {
+        $number = $this->mixedToPhoneNumber($value);
+
+        return $number ? $number->toE164() : $value;
```

Comment:
Does this filter actually filter? Should it really ignore non-phone-numbers? :thinking:

Reply:
The way most of the laminas filters are implemented, appear to me, to always attempt to filter a mixed value if possible or return the input if not - exceptions are only thrown when there's a configuration problem. I'm sure there are exceptions to the rule but I just scanned through a bunch of standard filters and they all seem to operate this way.